cmake/gcovr.cmake

```diff
@@ -1,16 +1,12 @@
-if(USE_GCOV)
- 
-    SET(GCOVR gcovr)
- 
-    SET(GCC_COVERAGE_COMPILE_FLAGS "-fprofile-arcs -ftest-coverage")
-    SET(GCC_COVERAGE_LINK_FLAGS "-lgcov")
-    SET( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${GCC_COVERAGE_COMPILE_FLAGS}" )
-    SET( CMAKE_EXE_LINKER_FLAGS "${CMAKE_LINKER_FLAGS} ${GCC_COVERAGE_LINK_FLAGS}" )
+if(BUILD_WITH_COVERAGE)
+
+    find_program (GCOVR gcovr)
```

Comment:
Thanks for the changes. 

The only missing piece would be to add a check after this line, something like:

```
if(NOT GCOVR)
  message(FATAL_ERROR "gcovr not found")
endif()
```

Reply:
Just added the change. Thank you very much for the help!

Reply:
@piponazo , there are some failures in the CI tests due to the lack of gcovr. If a user doesn't have it, should we just not apply the target?

Reply:
> @piponazo , there are some failures in the CI tests due to the lack of gcovr. If a user doesn't have it, should we just not apply the target?

I agree with that suggestion: gcovr is not essential for obtaining the test coverage data on the CI.

Reply:
We've changed the if statement to `if(GCOVR)`, and put the flags inside the if statement. That should solve the CI problem.

Reply:
Fine for me as it is! Thank you a lot for the contribution 😄

Reply:
Thank you! We greatly enjoyed this experience!